packages/swing-store/src/transcriptStore.js

```diff
@@ -235,6 +304,106 @@ export function makeTranscriptStore(
     return bounds;
   }
 
+  const sqlGetSpanEndPos = db.prepare(`
+    SELECT endPos
+    FROM transcriptSpans
+    WHERE vatID = ? AND startPos = ?
+  `);
+  sqlGetSpanEndPos.pluck(true);
+
+  const sqlReadSpanItems = db.prepare(`
+    SELECT item
+    FROM transcriptItems
+    WHERE vatID = ? AND ? <= position AND position < ?
+    ORDER BY position
+  `);
+
+  /**
+   * Read the items in a transcript span
+   *
+   * @param {string} vatID  The vat whose transcript is being read
+   * @param {number} [startPos] A start position identifying the span to be
+   *    read; defaults to the current span, whatever it is
+   *
+   * @returns {IterableIterator<string>}  An iterator over the items in the indicated span
+   */
+  function readSpan(vatID, startPos) {
```

Comment:
Review nit: it would make the life of the reviewer a lot easier if there was a separate commit just moving the original implementation around first.

Reply:
I can understand why that would be nice but it's not really practical.  These changes happen due to implementation changes introducing new forward references, which eslint then whines about until you either put in an override pragma or move the code.  Since each such movement thus happens at a different point in the implementation process, it's not straightforward to rearrange them into their own commit.